src/variableProcessor.ts

```diff
@@ -40,9 +45,124 @@ export class VariableProcessor {
             }
         }
 
+        // any vars that make decisions on the URL (request var) must be last so all vars have been evaluated
+        let aadRegex = new RegExp(aadRegexPattern, 'g');
+        if (aadRegex.test(request)) {
+            request = request.replace(aadRegex, await VariableProcessor.getAadToken(request));
+        }
+
         return request;
     }
 
+    public static async clearAadTokenCache() {

```

Comment:
If we want this method to be invoked by VS Code command, we need to update the __extension.ts__ and __package.json__ too.

Reply:
Well, I kept this around because I didn't want to add the command right now, but I didn't realize how easy it was going to be :-P Thanks for the pointer!